src/SQLStore/ByIdDataRebuildDispatcher.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+namespace SMW\SQLStore;
+
+use SMW\SemanticData;
+use SMM\DIWikiPage;
+use SMW\DIProperty;
+use SMW\Store;
+use SMW\MediaWiki\Jobs\UpdateJob;
+use SMW\MediaWiki\Jobs\JobBase;
+use Onoi\MessageReporter\MessageReporter;
+
+use Title;
+use Hooks;
+
+/**
+ * @license GNU GPL v2+
+ * @since 2.3
+ *
+ * @author Markus Krötzsch
+ * @author Jeroen De Dauw
+ * @author Nischay Nahata
+ * @author mwjames
+ */
+class ByIdDataRebuildDispatcher implements MessageReporter {
+
+	/**
+	 * @var Store
+	 */
+	private $store = null;
+
+	/**
+	 * @var boolean
+	 */
+	private $useUpdateJobParseMode = true;
+
+	/**
+	 * @var boolean
+	 */
+	private $useJobQueue = true;
+
+	/**
+	 * @var array|false
+	 */
+	private $namespaces = false;
+
+	/**
+	 * @var integer
+	 */
+	private $iterationLimit = 1;
+
+	/**
+	 * @since 2.3
+	 *
+	 * @param Store $store
+	 */
+	public function __construct( Store $store ) {
+		$this->store = $store;
+	}
+
+	/**
+	 * @see MessageReporter::reportMessage
+	 *
+	 * @since 2.3
+	 *
+	 * @param string $message
+	 */
+	public function reportMessage( $message ) {
```

Comment:
I'm wondering why you implemented `MessageReporter`. This is certainly an odd implementation. If you want to do a no-op, then why return a string?